src/graphics/program-lib/chunks/msdf.frag

```diff
@@ -35,17 +35,24 @@ vec4 applyMsdf(vec4 color) {
     float sigDist = median(tsample.r, tsample.g, tsample.b);
     float sigDistShdw = median(ssample.r, ssample.g, ssample.b);
 
+    float tsamplea = texture2D(texture_msdfMap, vUv0).a;
+    float ssamplea = texture2D(texture_msdfMap, uvShdw).a;
+
+    sigDist=mix(sigDist,tsamplea, clamp((0.4-tsamplea)*10.0,0.0,1.0));
+    sigDistShdw=mix(sigDistShdw,ssamplea, clamp((0.4-ssamplea)*10.0,0.0,1.0));
+
     #ifdef USE_FWIDTH
         // smoothing depends on size of texture on screen
         vec2 w = fwidth(vUv0);
-        float smoothing = clamp(w.x * font_textureWidth / font_pxrange, 0.0, 0.5);
+        float smoothing = 0.0; // smoothing doesn't work with multi-pass stencil rendering
+        //float smoothing = clamp(w.x * font_textureWidth / font_pxrange, 0.0, 0.5);
     #else
         float font_size = 16.0; // TODO fix this
         // smoothing gets smaller as the font size gets bigger
         // don't have fwidth we can approximate from font size, this doesn't account for scaling
         // so a big font scaled down will be wrong...
-
-        float smoothing = clamp(font_pxrange / font_size, 0.0, 0.5);
+        float smoothing = 0.0; // smoothing doesn't work with multi-pass stencil rendering
+        //float smoothing = clamp(font_pxrange / font_size, 0.0, 0.5); 
```

Comment:
Slightly concerned about this. Does setting smoothing to 0 make edges look noticeably worse? Do you start seeing nasty aliasing when the text is reasonably small?

Reply:
I did a comparison with smaller text with and without smoothing and it does look more aliased when using stencils without smoothing - unacceptably so IMO - but unfortunately, smoothing doesn't work with stenciling - so we will have to think up another solution that works with smoothing/anti-aliasing.

Reply:
I think I can just apply the smoothing in the alpha - I will try it

Reply:
Code updated to now work with smoothing! :)